server/src-lib/Hasura/RQL/Types/Common.hs

```diff
@@ -277,12 +278,26 @@ isSystemDefined = unSystemDefined
 successMsg :: EncJSON
 successMsg = "{\"message\":\"success\"}"
 
+newtype NonNegativeInt = NonNegativeInt { getNonNegativeInt :: Int }
+  deriving (Show, Eq, ToJSON, Generic, NFData, Cacheable, Num, Read)
+
+maybeInt :: MonadFail m => Maybe Int -> m Int
+maybeInt x = case x of
+  Just v  -> return v
+  Nothing -> fail "integer passed is out of bounds"
```

Comment:
Can move this function to the `FromJSON` instance of `NonNegativeInt`, since this function is used only once.